### PR TITLE
[Swift] Use llvm casts instead of looking at {Decl,TypeBase}::getKind()

### PR DESCRIPTION
**Pull request by slavapestov**

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test Linux

**Comment by slavapestov** [comment]
@swift-ci Please test Linux

**Comment by slavapestov** [comment]
@swift-ci Please test Linux

**Comment by slavapestov** [comment]
@swift-ci Please test Linux

**Comment by slavapestov** [comment]
@swift-ci Please test Linux

**Comment by slavapestov** [comment]
@swift-ci Please test Linux

File: source/Symbol/SwiftASTContext.cpp [hunk 1]
```diff
@@ -494,7 +494,7 @@ CachedMemberInfo *SwiftASTContext::GetCachedMemberInfo(void *type) {
 
           for (auto decl : t_decl->getMembers()) {
             // Find ivars that aren't properties
-            if (decl->getKind() == swift::DeclKind::Var) {
+            if (swift::isa<swift::VarDecl>(decl)) {
```
Comment (jrose-apple) [inline source/Symbol/SwiftASTContext.cpp]: `isa` actually comes from `llvm::`, not `swift::`.

**Pull request merged**